mitmproxy/addons/tlsconfig.py

```diff
@@ -126,17 +128,17 @@ def create_client_proxy_ssl_conn(self, tls_start: tls.TlsStartData) -> None:
             min_version=net_tls.Version[ctx.options.tls_version_client_min],
             max_version=net_tls.Version[ctx.options.tls_version_client_max],
             cipher_list=cipher_list,
-            cert=cert,
-            key=key,
-            chain_file=chain_file,
+            cert=entry.cert,
+            key=entry.privatekey,
+            chain_file=entry.chain_file,
             request_client_cert=False,
             alpn_select_callback=alpn_select_callback,
             extra_chain_certs=server.certificate_list,
             dhparams=self.certstore.dhparams,
         )
         tls_start.ssl_conn = SSL.Connection(ssl_ctx)
         tls_start.ssl_conn.set_app_data(AppData(
-            server_alpn=server.alpn,
+            server_alpn=always_bytes(server.alpn, "utf8", "replace"),
```

Comment:
Technically ALPN does not say anything about UTF-8 - just a plain byte string that is registered with IANA.
`replace` in this context sounds wrong/dangerous - but very unlikely to cause a problem in the real world.

Reply:
Fixed. 👍